### PR TITLE
More utils

### PR DESCRIPTION
**Pull request by FG-TUM**
# Description

Free functions for parsing typical autopas options from strings to enums since this might / will be used by users (already required in ls1).

- [x] parseTraversalOptions
- [x] parseContainerOptions
- [x] parseSelectorStrategy
- [x] parseDataLayout
- [x] robust parsing
- [x] to_string() for all enums

## ~Resolved Issues~

# How Has This Been Tested?

- [x] played around with it in md-flex (does this really require unit-tests? :innocent:)
- [x] ok I ended up writing templated functional tests....

### CODE REVIEW

**Comment by FG-TUM** [comment]
> good so far, but there should be some function returning possible entries or at least an exception if no matching string was found.

Output via logging -> No since we cannot assume that the logger is initialized.
Exceptions -> No since they might need the logger.
Output via cout/cerr -> Nope!

Possibilities are stated in the documentation.
If nothing was found an empty string or error value is returned.

**Comment by SteffenSeckler** [comment]
hmm, how about an additional function returning a vector of strings of possible values, such that it can be machine parsed?

File: tests/testAutopas/tests/utils/StringParserTest.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * @param allOptions Vector of all expected options.
+ * @param optionsString String that shall be parsed.
+ * @param parseFun Parsing Function.
+ */
+template <class T>
+void testParseMultiple(const std::vector<T> &allOptions, const std::string &optionsString,
+                       std::function<std::vector<T>(const std::string &)> &&parseFun) {
+  auto parsedOptions = parseFun(optionsString);
+
+  std::sort(parsedOptions.begin(), parsedOptions.end());
```
[the remark this answers is not in the document]
Comment (FG-TUM) [reply tests/testAutopas/tests/utils/StringParserTest.h]: a08e92b

File: src/autopas/utils/StringParser.h [hunk 2]
```diff
@@ -19,66 +19,110 @@ namespace utils {
...
+
+  return wordVector;
+}
+
 /**
  * Converts a string of options to a vector of enums. The options are expected to be lower case.
+ *
+ * Possible options: c01, c08, c18, direct, sliced, v01, v18, vsl
+ *
  * @param traversalOptionsString String containing traversal options.
```
[the remark this answers is not in the document]
Comment (FG-TUM) [reply src/autopas/utils/StringParser.h]: they literally document themselves in the doxygen doc.

File: src/autopas/utils/StringParser.h [hunk 3]
```diff
@@ -19,66 +19,110 @@ namespace utils {
...
+  }
+  if (prev < searchString.length()) wordVector.push_back(searchString.substr(prev, std::string::npos));
+
+  return wordVector;
+}
+
 /**
  * Converts a string of options to a vector of enums. The options are expected to be lower case.
+ *
+ * Possible options: c01, c08, c18, direct, sliced, v01, v18, vsl
```
[the remark this answers is not in the document]
Comment (FG-TUM) [reply src/autopas/utils/StringParser.h]: 0e058782369c4a47c4e1af1d1275d93da245b482

**Pull request merged**